Adds basic character movement

oliverydapper opened the pull request:
Jumping, walking, turning, all implemented and all in CharacterController script.

oliverydapper committed:
Implements left and right movement and collision detection

Left + right movement enabled without reversing direction sprite is pointing, collision with platform implemented, no jumping yet

oliverydapper committed:
Merge branch 'ala' into Jordan

oliverydapper committed:
Combines animations with left/right movement, jumping

Still do not have sprite image flipping when reversing direction, but walking, idle, and jump work

oliverydapper committed:
Adds penguin sprite flip

Penguin sprite flips back and forth when moving left and right

oliverydapper committed:
Merge branch 'main' into Jordan

oliverydapper committed:
Resolving conflicts

hopefully resolves merge conflicts (in TideyUp.sln)

P-Wallis commented on CharacterController.cs:
```diff
@@ -0,0 +1,76 @@
+            velocity.y += delta * gravity;
+        }
+		
+        if (Input.IsActionPressed("ui_left") && direction=="left")
```
Instead of 
```
if (Input.IsActionPressed("ui_left") && direction=="left")
{
    velocity.x = -walkSpeed;
    _animatedSprite.Play("Run");
}
else if(Input.IsActionPressed("ui_left") && direction == "right")
{
    direction = "left";
    _animatedSprite.Scale = new Vector2(-.3f, .3f);
    velocity.x = -walkSpeed;
    _animatedSprite.Play("Run");
}
```


Try
```
if(Input.IsActionPressed("ui_left"))
{
    if(direction == "right")
    {
        direction = "left";
        _animatedSprite.Scale = new Vector2(-.3f, .3f);
    }
    velocity.x = -walkSpeed;
    _animatedSprite.Play("Run");
}
```

P-Wallis commented on CharacterController.cs:
```diff
@@ -0,0 +1,76 @@
+    public int JumpImpulse =  -400;
+    Vector2 velocity;
+    public AnimatedSprite _animatedSprite;
+    public string direction = "right";
```
Consider making this an enum!

oliverydapper committed:
Fixes Peter's code review issues

-Reformats if/else logic in physics process in character controller
-uses enums to change direction instead of strings
-saves sprite scale as a variable so not hard coded
-removes unused Node2d.tscn

P-Wallis reviewed (approved):
Looks good!
